exonum-java-binding/core/rust/src/utils/resource_manager/imp.rs

```diff
@@ -160,111 +189,129 @@ mod tests {
     use std::i64;
 
     enum T {}
+    enum HandleFooType {}
+    enum HandleBarType {}
+    const HANDLE_FOO: Handle = 1000;
+    const HANDLE_BAR: Handle = 2000;
     const INVALID_HANDLE: Handle = i64::MAX;
-
-    // Unique ("valid") handles should be used in the each test because `HANDLES_MAP` is a shared
-    // state and tests are run concurrently.
-    const MANAGE_HANDLES_FIRST_HANDLE: Handle = 1000;
-    const MANAGE_HANDLES_SECOND_HANDLE: Handle = 2000;
-    const MANAGE_HANDLES_NON_OWNED_HANDLE: Handle = 3000;
-    const DUPLICATED_HANDLE: Handle = 4000;
-    const WRONG_TYPE_HANDLE: Handle = 5000;
-    const WRONG_OWNERSHIP_HANDLE: Handle = 6000;
+    const ZERO_HANDLE: Handle = 0;
 
     #[test]
-    fn manage_handles() {
-        // Add Java-owned handle.
-        enum T1 {}
-        add_handle::<T1>(MANAGE_HANDLES_FIRST_HANDLE);
-        check_handle::<T1>(MANAGE_HANDLES_FIRST_HANDLE);
-        check_handle_impl::<T1>(
-            MANAGE_HANDLES_FIRST_HANDLE,
-            Some(HandleOwnershipType::JavaOwned),
+    fn check_handle_type_and_ownership_ok() {
+        let handle_info = prepare_handle_info();
+        HandlesStorage::check_handle_type_and_ownership::<HandleFooType>(
```

Comment:
Do we need tests using this internal static method? Wouldn't it be simpler to 
1. Create a fresh HandleStorage.
2. Setup the test data using public API.
3. Perform the action we'd like to test.
4. Verify.
?